### DISCUSSION TITLE
How to extract an integer from an array obtained via jsonpath query?

### DISCUSSION THREAD

**Discussion by meleu-cw** [Orange-OpenSource/hurl#2642] [Q&A] [answered]
I have a situation where I capture the `id` with a jsonpath query in a response and try to use the `id` in the URL of the next request.

The issue is that the `id` comes in an array, like `[1234]`, and I don't know how to extract the number out of this array notation.

Here's an example of the JSON response:
```json
[
    {
        "id": 5954,
        "status": "executed",
        // ...
    },
    {
        "id": 5951,
        "status": "repaid",
        // ...
    },
    // more items...
]
```

Here's the jsonpath I'm using: `$.[?(@.status == 'executed')].id`, which results in `[1234]` (while I want `1234`).

> **Note**: as it's an array, I tried appending `[0]` at the end of my query. But it results in this error:
> 
> ```
> loan_id: jsonpath "$.[?(@.status == 'executed')].id[0]"
>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ The query didn't return any result
> ```

Here's the hurl file I'm using:

```
GET {{BASE_URL}}/loans
HTTP 200
[Captures]
loan_id: jsonpath "$.[?(@.status == 'executed')].id"

# repayments
GET {{BASE_URL}}/loans/{{loan_id}}/repayments
HTTP 200
```

When I run it, I have this error:
```
error: Unrenderable variable
  --> file.hurl:17:43
   |
17 | GET {{BASE_URL}}/loans/{{loan_id}}/repayments
   |                          ^^^^^^^ variable <loan_id> with value [5954] can not be rendered
   |
```

How can I solve this?

**Comment by jcamiel** [comment, marked as answer]
Hi,

Your JSONPath is correct. As you've got an array, you can refine your query with [filters](https://hurl.dev/docs/filters.html). The filter `nth` can be used in this case:

```hurl
GET {{BASE_URL}}/loans
HTTP 200
[Captures]
loan_id: jsonpath "$.[?(@.status == 'executed')].id" nth 0
```

That should be good now!

**Reply by meleu-cw** [reply]
yeah, the `nth` filter works beautifully!
thanks!
